src/main/java/org/thingsboard/gateway/extensions/sigfox/conf/mapping/SigfoxDeviceMapping.java

```diff
@@ -0,0 +1,110 @@
+        try {
+            DocumentContext document = JsonPath.parse(body);
+            long ts = System.currentTimeMillis();
+            String deviceName = eval(document, deviceNameJsonExpression);
+            if (!StringUtils.isEmpty(deviceName)) {
+                List<KvEntry> attrData = getKvEntries(JsonPath.parse(body), attributes);
+                List<TsKvEntry> tsData = getKvEntries(JsonPath.parse(body), timeseries).stream()
+                        .map(kv -> new BasicTsKvEntry(ts, kv))
+                        .collect(Collectors.toList());
+                return new DeviceData(deviceName, attrData, tsData);
+            }
+        } catch (Exception e) {
+            log.error("Exception occurred while parsing json request body [{}]", body, e);
+            throw e;
+        }
+        return null;
+    }
+
+    private List<KvEntry> getKvEntries(DocumentContext document, List<? extends SigfoxKVMapping> mappings) {
+        List<KvEntry> result = new ArrayList<>();
+        if (mappings != null) {
+            for (SigfoxKVMapping mapping : mappings) {
+                String key = eval(document, mapping.getKey());
+                String strVal = eval(document, mapping.getValue());
+                switch (mapping.getType().getDataType()) {
+                    case STRING:
+                        result.add(new StringDataEntry(key,
+                                applyTransformer((DataValueTransformer<String>) mapping.getTransformer(), strVal)));
+                        break;
+                    case BOOLEAN:
+                        result.add(new BooleanDataEntry(key,
+                                applyTransformer((DataValueTransformer<Boolean>) mapping.getTransformer(), Boolean.valueOf(strVal))));
+                        break;
+                    case DOUBLE:
+                        result.add(new DoubleDataEntry(key,
+                                applyTransformer((DataValueTransformer<Double>) mapping.getTransformer(), Double.valueOf(strVal))));
+                        break;
+                    case LONG:
+                        result.add(new LongDataEntry(key,
+                                applyTransformer((DataValueTransformer<Long>) mapping.getTransformer(), Long.valueOf(strVal))));
+                        break;
+                }
+            }
+        }
+        return result;
+    }
+
+    private static <T> T applyTransformer(DataValueTransformer<T> transformer, T value) {
+        if (transformer != null) {
+            try {
+                return transformer.transform(value);
+            } catch (Exception e) {
+                log.error("Transformer [{}] can't be applied to field with key [{}] and value [{}]",
+                        transformer.getName(), value);
+                throw e;
+            }
+        } else {
+            return value;
+        }
+    }
+
+    protected static String eval(DocumentContext document, String expression) {
```

Comment:
you can just extend AbstractJsonConverter and this method will be available, right?